great_expectations/checkpoint/checkpoint.py

```diff
@@ -43,49 +47,80 @@ def __init__(
         runtime_configuration: Optional[dict] = None,
         validations: Optional[List[dict]] = None,
         profilers: Optional[List[dict]] = None,
-        # Next two fields are for LegacyCheckpoint configuration
         validation_operator_name: Optional[str] = None,
         batches: Optional[List[dict]] = None,
     ):
         self._name = name
+        # Note the gross typechecking to avoid a circular import
+        if "DataContext" not in str(type(data_context)):
+            raise TypeError("A checkpoint requires a valid DataContext")
         self._data_context = data_context
 
-        checkpoint_config: CheckpointConfig = CheckpointConfig(
-            **{
-                "name": name,
-                "config_version": config_version,
-                "template_name": template_name,
-                "module_name": module_name,
-                "class_name": class_name,
-                "run_name_template": run_name_template,
-                "expectation_suite_name": expectation_suite_name,
-                "batch_request": batch_request,
-                "action_list": action_list,
-                "evaluation_parameters": evaluation_parameters,
-                "runtime_configuration": runtime_configuration,
-                "validations": validations,
-                "profilers": profilers,
-                # Next two fields are for LegacyCheckpoint configuration
-                "validation_operator_name": validation_operator_name,
-                "batches": batches,
-            }
-        )
+        if configurator:
+            configurator_obj = configurator(
+                **{
+                    "class_name": configurator.get("class_name"),
+                    "name": name,
+                    "data_context": data_context,
+                    "config_version": config_version,
+                    "template_name": template_name,
+                    "class_name": class_name,
+                    "module_name": module_name,
+                    "run_name_template": run_name_template,
+                    "expectation_suite_name": expectation_suite_name,
+                    "batch_request": batch_request,
+                    "action_list": action_list,
+                    "evaluation_parameters": evaluation_parameters,
+                    "runtime_configuration": runtime_configuration,
+                    "validations": validations,
+                    "profilers": profilers,
+                    # Next two fields are for LegacyCheckpoint configuration
+                    "validation_operator_name": validation_operator_name,
+                    "batches": batches,
+                },
```

Comment:
conceptually this makes sense but syntactically shouldn't this just be parameters to the constructor (i.e. name=foo,...)

Reply:
...and the configurator should also get `**kwargs`

Reply:
Yeah, I just did dict unpacking since I already had it. Will change.